rust/publish-zenodo/src/lib.rs

```diff
@@ -0,0 +1,575 @@
+
+        // Create the SWB bundle
+        SwbCodec::default()
+            .to_path(
+                &root,
+                &swb_path, 
+                Some(EncodeOptions {
+                    format: Some(Format::Swb),
+                    standalone: Some(true),
+                    ..Default::default()
+                }),
+            )
+            .await?;
+
+        if self.dry_run {
+            tracing::info!("Dry run complete - bundle created at {}", swb_path.display());
+            return Ok(());
+        }
+
+        // Get bucket URL for file upload
+        let bucket_url = deposition["links"]["bucket"]
+            .as_str()
+            .ok_or_eyre("No bucket URL in response")?;
+
+        // Upload the SWB file using same filename
+        let file_name = format!("{}.swb", filename);
+        let url: String = format!("{}/{}", bucket_url, file_name);
+        tracing::info!(url = url, file = file_name, "Uploading deposit");
+        let upload_response = client
+            .put(&format!("{}/{}", bucket_url, file_name))
+            .query(&[("access_token", &token)])
+            .body(tokio::fs::read(&swb_path).await?)
+            .send()
+            .await?;
+
+        if !upload_response.status().is_success() {
+            tracing::error!(response = ?upload_response, file = ?file_name, "Failed to upload");
+            bail!("Failed to upload file: {}", upload_response.text().await?);
+        }
+
+        let deposition_url = deposition["links"]["html"].as_str().ok_or_eyre("No deposit URL provided by Zenodo.")?;
+        
+        if self.force {
+            // Publish the deposition
+            let publish_response = client
+                .post(format!(
+                    "{}/api/deposit/depositions/{}/actions/publish",
+                    server_url, deposition_id
+                ))
+                .query(&[("access_token", &token)])
+                .send()
+                .await?;
+
+            if !publish_response.status().is_success() {
+                bail!("Failed to publish deposition: {}", publish_response.text().await?);
+            }
+
+            tracing::info!("Created published deposition at {deposition_url}.");
+        } else {
+
+            let mut msg = format!("Successfully published document as a draft deposit to {deposition_url}");
+            if let Some(doi) = reserved_doi {
+                msg.push_str(&format!(" with the doi:{doi} pre-reserved"));
+            }
+            msg.push('.');
+            println!("{msg}");
+            println!("Review and publish manually, or use --force to publish directly");
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod cli_tests {
+    use super::*;
+
+    #[test]
+    fn defaults_to_sandbox() {
+        let args = Cli::parse_from(&["publish-zenodo"]);
+        assert!(args.sandbox);
+    }
+
+    /// Tests that the publication type defaults to pre-print when --publication is set
```

Comment:
for the Zenodo vocab it should be `preprint` not `pre-print`

Reply:
Corrected.